src/app/new-movie-modal/new-movie-modal.component.ts

```diff
@@ -11,26 +10,32 @@ import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
 })
 export class NewMovieModalComponent implements OnInit {
 
-  movieForm: FormGroup
+  form: FormGroup
 
-  constructor(public activeModal: NgbActiveModal, private movieService: MovieService, 
-              private formBuilder: FormBuilder) { 
-    this.movieForm = this.formBuilder.group({
-      title: [''],
-      posterUrl: [''],
-      genre: [''],
-      rating:['']
-      })
-    }
+  constructor(
+    public activeModal: NgbActiveModal,
+    private movieService: MovieService,
+    private formBuilder: FormBuilder
+  ) {
+
+    this.form = this.formBuilder.group({
+      title: ['', [Validators.required]],
+      posterUrl: ['', [Validators.required]],
+      genre: ['', [Validators.required]]
+    });
+
+  }
 
   ngOnInit(): void {
   }
 
-  submitNewMovieModal() {
-   // onSubmitNewMovie(){
-      this.movieService.addMovie(this.movieForm.value).subscribe(() => {
-      this.activeModal.close();
-      window.location.reload();
-    });
```

Comment:
Why are we not using the subscribe method?

Reply:
We changed the method to return a `Promise` instead of an `Observable` since we thought creating a movie was a one-time thing.